### ISSUE TITLE
newline-per-chained-call prop ignoreChainWithDepth not working as expected

### ISSUE THREAD

**Issue by Fabioni** [eslint-community/eslint-stylistic#366]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-community/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

This bug from the original eslint repo, is still appearing in @stylistics, I am not sure why.

> With the ignoreChainWithDepth value, there's no way to say "if the chain depth is no longer ignored, all values need newlines".
> At the point that this rule is triggered, instead of allowing only the first chained prop to stay at the top, all props would be on a new line.

I'll save further the explanation and just refer you to the other issue: https://github.com/eslint/eslint/issues/12970

I do not get the idea of the minimal reproduction of Stackblitz. When I want to save the file, it automatically reformates the code so the problem I want to show can not be shown...

So I copy&pase it here:

```js
const mypromise = new Promise(() => {});

mypromise.then(() => {
    console.log('foo');
  })
  .catch(() => {
    console.log('bar');
  });
```

should be 

```js
const mypromise = new Promise(() => {});

mypromise
  .then(() => {
    console.log('foo');
  })
  .catch(() => {
    console.log('bar');
  });

```

### Reproduction

https://stackblitz.com/edit/github-4jgvhh-bvywmk?file=src%2Findex.ts

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
